### ISSUE TITLE
Allow user to optionally define their overall coin profile by toggling individual tiers.

### ISSUE THREAD

**Issue by emergent-reasons** [clifordsymack/Electron-Cash#88]
There is a lot of discussion about enabling high tiers for businesses/whales. I think this proposal will provide a good UX for whales as well as all other types of users.

## Toggle-able tiers

- The shuffle client provides toggle-able tiers with a sane default, and most users never need to touch it.
- Whales can expand their envelope up by enabling higher tiers
- Users unwilling to mix coins bigger than `x` can disable all tiers over their limit. `x` could vary wildly depending on wealth and risk tolerance.
- users that don't want to grind below `y` can disable the low tiers
- etc.

## Immediate benefit:

It should resolve all the specific concerns I have heard so far:

- want higher tiers
- don't want lower tiers
- shuffling must be uncomplicated and unsurprising for the user who never touches settings

## Future shuffle-up benefit:

when shuffle-up is implemented, the profile becomes even more useful. It becomes a declaration of what kind of coins you want and the client can shuffle up/down to meet that.

## Future preset profiles benefit:

This is questionable but possible in the future if all of the above works well. Some apps use preset profiles that cover a specific target group. For example:

- default
- miner (enable higher tiers, disable lower ones)
- conservative (nothing over `x`)
- Every Day I'm Shuffling (all tiers)
- fiat-oriented (your tier window shifts depending on the market price)
- custom
- ... whatever makes sense based on actual usage

**Comment by cculianu** [comment]
I think this proposal is perfect.  It even naturally inspires me to compose a settings dialog of sorts for it.

I am going to go ahead and start on that now.  I want to get this in before Jonald declares an absolute feature freeze. :)

**Comment by cculianu** [comment]
I think the initial dialog will be very simple with just checkboxes for all the tiers plus a low and high end coin cutoff you can set on top of that.

**Comment by emergent-reasons** [comment]
Great! What's the benefit of the additional cutoff? Maybe sometimes 10x is too big of a gap?

**Comment by cculianu** [comment]
Yeah sometimes 10x might be too big.  Consider 100 BCH -> 1000 BCH etc.

Ok so I implemented a simplified version of this -- here is the pr: #90 .

I don't think I will do anything more sophisticated now -- but your suggestion is good and will leave this issue open.

**Comment by emergent-reasons** [comment]
:+1: The min/max is nice to have. I think many will thank you for adding it.

I want to add an extra note since there is a min/max setting now. The interaction of "min/max coin" and shuffle scales is non-trivial. Therefore if the selectable tiers are ever implemented, the tiers and min/max need to interact and avoid conflict with each other so that no surprising behavior happens. E.g. the user should not be able to set a minimum coin above their lowest tier and the reverse also.

I'm shit at design but I think the most intuitive layout would have the tiers with the max over the top tier and the min under the bottom tier.

```text
 ______ biggest coin to shuffle
☐ 10
☑ 1
☑ .1
☑ .01
☑ .001
☐ .0001
 ______ smallest coin to shuffle
```

or maybe more noisy/explicit:

```text
 ______ biggest coin to shuffle
☐ 100   ~ 1000-
☐ 10    ~ 100-
☑ 1     ~ 10-
☑ .1    ~ 1-
☑ .01   ~ .1-
☑ .001  ~ .01-
☐ .0001 ~ .001-
 ______ smallest coin to shuffle
```
or the same with number ranges stacked vertically instead of listed horizontally.

```text
<insert great design here>
```

**Comment by cculianu** [comment]
I agree with this scheme -- definitely if they select tier above their minimum, the minimum should auto-populate in the UI to reflect that, then pop back when they unselect it.  I agree.

I like your use of unicode here :)
